src/script.c

```diff
@@ -430,14 +430,24 @@ static int scriptVerifyClusterState(scriptRunCtx *run_ctx, client *c, client *or
     c->flags |= original_c->flags & (CLIENT_READONLY | CLIENT_ASKING);
     int hashslot = -1;
     if (getNodeByQuery(c, c->cmd, c->argv, c->argc, &hashslot, &error_code) != server.cluster->myself) {
-        if (error_code == CLUSTER_REDIR_DOWN_RO_STATE) {
-            *err = sdsnew(
-                    "Script attempted to execute a write command while the "
-                            "cluster is down and readonly");
+        if (error_code == CLUSTER_REDIR_CROSS_SLOT) {
+            *err = sdsnew("Script attempted to access keys don't hash to the same slot");
```

Comment:
It is better to differentiate from error without SCRIPT_ALLOW_CROSS_SLOT, CLUSTER_REDIR_CROSS_SLOT is generated by command, not by script.

Maybe "Command %s in script attempted to access keys don't hash to the same slot"